src/main/java/org/wildfly/swarm/plugin/process/ModuleFiller.java

```diff
@@ -108,41 +119,49 @@ public FractionMetadata apply(FractionMetadata meta) {
 
             locateFillModules(potentialModules, requiredModules, availableModules);
 
+            DecimalFormat fmt = new DecimalFormat("###0.00", new DecimalFormatSymbols(Locale.US));
+            double bytesInMegabyte = 1024.0 * 1024.0;
+
             long size = 0;
-            DecimalFormat fmt = new DecimalFormat("####.00");
+            boolean unknownSize = false;
             for (Artifact artifact : this.allArtifacts) {
                 ArtifactRequest req = new ArtifactRequest();
                 req.setArtifact(artifact);
 
+                String artifactSizeStr = "???";
                 try {
                     ArtifactResult artifactResult = this.resolver.resolveArtifact(repositorySystemSession, req);
                     if (artifactResult.isResolved()) {
                         File file = artifactResult.getArtifact().getFile();
                         long artifactSize = Files.size(file.toPath());
                         size += artifactSize;
-                        this.log.info(String.format("%100s %10s mb", toModuleArtifactName(artifact), fmt.format(artifactSize / (1024.0 * 1024.0))));
+                        artifactSizeStr = fmt.format(artifactSize / bytesInMegabyte);
+                    } else {
+                        unknownSize = true;
                     }
-                } catch (Exception e) {
-                    this.log.error(e.getMessage(), e);
+                } catch (ArtifactResolutionException | IOException e) {
+                    unknownSize = true;
                 }
+                this.log.info(String.format("%100s %10s MB", toModuleArtifactName(artifact), artifactSizeStr));
             }
-            this.log.info(this.project.getArtifactId() + ": total size:  " + fmt.format(size / (1024.0 * 1024.0)) + " mb");
+            String sizeStr = fmt.format(size / bytesInMegabyte);
+            this.log.info(this.project.getArtifactId() + ": total size:  " + sizeStr + (unknownSize ? "+" : "") + " MB");
```

Comment:
This will log "totalSize: 0+ MB" in case of the unknown size. Maybe "totalSize: unknown" ? '0+' is a bit cryptic unless it is a convention I'm unaware of...

Reply:
Fixed.